lib/workers/repository/process/lookup/index.ts

```diff
@@ -372,6 +372,40 @@ export async function lookupUpdates(
       );
       let shrinkedViaVulnerability = false;
       if (config.isVulnerabilityAlert) {
+        if (config.vulnerabilityFixVersion) {
+          res.vulnerabilityFixVersion = config.vulnerabilityFixVersion;
+          if (versioning.isVersion(config.vulnerabilityFixVersion)) {
+            // Filter out versions if the vulnerabilityFixVersion is higher
+            const fixedFilteredReleases = filteredReleases.filter(
+              (r) =>
+                !versioning.isGreaterThan(
```

Comment:
I understand that this works under the assumption `config.vulnerabilityFixVerison` is set as an exact version. For GitHub alerts, `first_patched_version` always includes an exact version. which was defined as a constraint in `allowedVersions` before the regression happened. Otherwise it would be this:
https://github.com/renovatebot/renovate/blob/ec509ab063f56c620093f507c4c30356ef3c8df7/lib/workers/repository/init/vulnerability.ts#L46-L51

For OSV, depending on the concrete advisory there are two kinds of fix versions possible, "first patched" and "last affected", which is why the constraint was either `>= fixVersion` or `> fixVersion`, respectively.

I'm afraid this approach could not distinguish between these two cases. How about making `vulnerabilityFixVersion` a constraint, analogous to `allowedVersions` and applying the same filtering procedure?

Reply:
I updated the code to allow both version or range